homeassistant/components/epson/media_player.py

```diff
@@ -133,7 +133,10 @@ async def async_update(self):
             self._source = SOURCE_LIST.get(source, self._source)
             volume = await self._projector.get_property(VOLUME)
             if volume:
-                self._volume = volume
+                try:
+                    self._volume = float(volume)
+                except ValueError:
+                    pass
```

Comment:
Instead of `pass`-ing, suppress it.
```suggestion
                with suppress(ValueError):
                    self._volume = float(volume)
```

However, I'm having trouble thinking this is the right solution. Shouldn't the state become "unknown" instead?

Reply:
Volume level can be float or None so I think this is the right solution https://github.com/home-assistant/core/blob/a91ca46342789861e310ac04e189430cc373a748/homeassistant/components/media_player/__init__.py#L477 , isn't it?

Reply:
Except that the volume isn't set to `None` when this occurs...

Reply:
by default it is set to None in media player entity in __init__ so I don't change value of it, right?

Reply:
Sure, but if it fails on the next update, it will remain there with the old value in the current logic.

Reply:
changed to self._volume = None if ValueError, can you check now?

Reply:
Yeah, that works. Thanks! 👍